src/lib/storage/reference_segment.hpp

```diff
@@ -35,6 +35,7 @@ class ReferenceSegment : public AbstractSegment {
 
   std::shared_ptr<AbstractSegment> copy_using_allocator(const PolymorphicAllocator<size_t>& /*alloc*/) const override;
 
+  NodeID get_numa_node_location() override;
```

Comment:
Kind of odd that `get` is specialized, but `set` is kept as is. In general, I was wondering if you really need the virtual getter and setter. Since the getter and setter are public, the member variable could just be public itself. The `ReferenceSegment` just sets the field at construction (and figures it out the same way you currently figure it out in this override). What do you think?